config/goglog/goglog.go

```diff
@@ -38,13 +42,47 @@ func newLogger() *LoggerType {
 			Hooks:     make(logrus.LevelHooks),
 			Level:     logrus.InfoLevel,
 		},
+		sentryHubs: make(map[sentry.Level]*sentry.Hub),
 	}
 }
 
-// LoggerType wrap logrus.Logger type
+// LoggerType wraps logrus.Logger type and adds a cache for Sentry hubs
 type LoggerType struct {
-	stdout *logrus.Logger
-	stderr *logrus.Logger
+	stdout     *logrus.Logger
+	stderr     *logrus.Logger
+	sentryHubs map[sentry.Level]*sentry.Hub
+	mu         sync.RWMutex
+}
+
+// Helper function to get or create a Sentry hub for a specific level
+func (t *LoggerType) getSentryHub(level sentry.Level) *sentry.Hub {
+	t.mu.RLock()
+	hub, ok := t.sentryHubs[level]
+	t.mu.RUnlock()
```

Comment:
Mutex will add extra execution cost, and the log levels are fixed, so it's better to initialize all hubs first.

```go
type LoggerType struct {
	stdout        *logrus.Logger
	stderr        *logrus.Logger
	sentryHubInfo *sentry.Hub
	sentryHubWarn *sentry.Hub
	// ...
}
```

Reply:
We have removed mutex, directly assiging hub to variable is not working for the second error so we have assigned function for Warn, fatal, Error

Reply:
```go

func newLogger() *LoggerType {
	warnHub := sentry.CurrentHub().Clone()
	warnHub.ConfigureScope(func(scope *sentry.Scope) {
		scope.SetLevel(sentry.LevelWarning)
	})
	// ...
}


// LoggerType wraps logrus.Logger type and adds a cache for Sentry hubs
type LoggerType struct {
	stdout         *logrus.Logger
	stderr         *logrus.Logger
	WarnSentryHub  *sentry.Hub
	// ...
}

func (t LoggerType) Warnf(format string, args ...any) {
	t.WarnSentryHub.CaptureMessage(fmt.Sprintf(format, args...))
	t.stdout.Warnf(format, args...)
}

func (t *LoggerType) Close() {
	sentry.CurrentHub().Flush(5 * time.Second)
	t.WarnSentryHub.Flush(5 * time.Second)
}
```

Reply:
@tsaikd can you review the changes now!